### PR TITLE
Add some emoji functions

### PR DESCRIPTION
**Pull request by NMW03**
**Please describe the changes this PR makes and why it should be merged:**
This change adding 2 function for check whether there is a unicode/custom emoji in the message content

**Status**
- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by NotSugden** [comment]
i would suggest at least attempting to follow the lint rules in future, i also don't personally see a need for these functions, perhaps something for the guide if anything really

File: src/util/Util.js [hunk 1]
```diff
@@ -573,6 +573,34 @@ class Util {
       setTimeout(resolve, ms);
     });
   }
+  
+  /**
+   * Checks whether there is a custom emoji in the message content
+   * @param {string} text to check
+   * @returns {Booelan}
+   */
+  static includesEmoji(text) {
```
Comment (Monbrey) [inline src/util/Util.js]: `Util#parseEmoji` already exists, literally in this file and does a better job

File: src/util/Util.js [hunk 2]
```diff
@@ -573,6 +573,34 @@ class Util {
...
+            return false
+        }
+    }
+  
+ /** 
+ * Check whether there is a unicode emoji in the message content
+ * @param {string} text to check
+ * @returns {Boolean}
+ */
+ static includesUnicodeEmoji(text) {
```
Comment (Monbrey) [inline src/util/Util.js]: Even if I thought this was a good method, which I don't, hardcoding this Unicode regex is a bad idea. It's not something this library should have to maintain.

*Theoretically* if implementing this in my own code, I'd propose using an external lib focused on unicode emoji to define this. I don't think discord.js should be doing this at all though as unlike custom emojis, this is not Discord related functionality.

**Comment by kyranet** [comment]
Duplicate of #3525, #1692, and #1691.

**Comment by SpaceEEC** [comment]
Personally -1 on adding (huge) regexes to the code that we won't even use ourselves or is related to other provided functionality.

File: src/util/Util.js [hunk 3]
```diff
@@ -573,6 +573,34 @@ class Util {
...
     });
   }
+  
+  /**
+   * Checks whether there is a custom emoji in the message content
+   * @param {string} text to check
+   * @returns {booelan}
+   */
+  static includesCustomEmoji(text) {
+        if (text !== String) throw new TypeError('Value is not a string')
```
Comment (Almeeida) [inline src/util/Util.js]: Perhaps this is what you were looking for: `typeof text !== 'string'`?

File: src/util/Util.js [hunk 4]
```diff
@@ -573,6 +573,34 @@ class Util {
...
+        }
+    }
+
+ /**
+ * Check whether there is a unicode emoji in the message content
+ * @param {string} text to check
+ * @returns {boolean}
+ */
+ static includesUnicodeEmoji(text) {
+        if (text !== String) throw new TypeError('Value is not a string')
```
Comment (Almeeida) [inline src/util/Util.js]:
```suggestion
        if (typeof text !== 'string') throw new TypeError('Value is not a string')
```

File: src/util/Util.js (hunk 3 above)
Comment (NotSugden) [inline src/util/Util.js]: constructing this error will result in throwing an error, as `TypeError` is defined as the custom djs error, see [here](https://github.com/discordjs/discord.js/blob/161f90761aa357f17066f4171a5f43f76a552eb7/src/util/Util.js#L5), [here](https://github.com/discordjs/discord.js/blob/161f90761aa357f17066f4171a5f43f76a552eb7/src/errors/DJSError.js#L59), [and here](https://github.com/discordjs/discord.js/blob/161f90761aa357f17066f4171a5f43f76a552eb7/src/errors/DJSError.js#L40), and you haven't provided a valid error message key, consider using the INVALID_TYPE error message key

File: src/util/Util.js (hunk 4 above)
Comment (NotSugden) [inline src/util/Util.js]: same as above

**Comment by iCrawl** [comment]
https://github.com/discordjs/discord.js/pull/3809#issuecomment-589951683

**Pull request closed without merging**